BDArmory/UI/BDATargetManager.cs

```diff
@@ -490,15 +490,21 @@ public static TargetSignatureData GetHeatTarget(Vessel sourceVessel, Vessel miss
                 // Abort if target is friendly.
                 if (mf != null)
                 {
-                    if (mf.Team.IsFriendly(tInfo.Team))
-                        continue;
+                    if (hasIFF)
+                    {
+                        if (mf.Team.IsFriendly(tInfo.Team))
+                            continue;
+                    }
                 }
 
                 // Abort if target is a missile that we've shot
                 if (tInfo.isMissile)
                 {
-                    if (tInfo.MissileBaseModule.SourceVessel == sourceVessel)
-                        continue;
+                    if (hasIFF)
+                    {
+                        if (tInfo.MissileBaseModule.SourceVessel == sourceVessel)
+                            continue;
+                    }
                 }
```

Comment:
The current code is the way it is due to years of people complaining about heaters homing in on friendlies. Why?
Additionally, this isn't going to do anything, due to later checks in the code to ensure the missile only locks to the designated target.